### PR TITLE
CUMULUS-1309 Build per-lambda deployment files in the API package

### PR DESCRIPTION
**Pull request by yjpa7145**
Fixes [CUMULUS-1309: Deploying lambdas in API package deploys entire API code in the lambda](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1309)

Before this PR, the entirety of the API package was being packaged into a single `index.js` file, and every lambda function was pointing at a different function inside that one 50 MB file.  This PR causes separate, Lambda function-specific, files to be generated and deployed.  These files now range anywhere from 350K to 8.1M.

The first step was to add separate entries for each function in the webpack config.  That caused webpack to generate an index.js for each lambda function in its own `packages/api/dist/[name]/`.  Once those separate, smaller files were generated, the `packages/api/config/*.yml` files were updated to point to those individual lambda function files.

Note: In some cases, the lambda function's was the default export from a module file.  That didn't work with the way that you need to specify the handler function in Lambda, so I had to change those cases from `module.exports = handler` to `module.exports = { handler }`.  These changes required updating some unit tests that invoked those handlers directly.

Lastly, after I verified that all of the tests were passing, I was able to check in CloudWatch logs to see how much memory was actually being used by the smaller Lambda functions, and adjusted the memory settings accordingly.  In a lot of cases, the memory allocation went from 768MB down to as low as 128MB.

### CODE REVIEW

File: example/spec/parallel/testAPI/snsRuleSpec.js [hunk 1]
```diff
@@ -64,6 +64,17 @@ describe('The SNS-type rule', () => {
 
   afterAll(async () => {
     await SNS.deleteTopic({ TopicArn: snsTopicArn }).promise();
+
```
Comment (yjpa7145) [inline example/spec/parallel/testAPI/snsRuleSpec.js]: The changes in this file are unrelated to this ticket, but Menno and I found that this test was not always cleaning up after itself, and would never pass afterward unless manual intervention was taken.  Made the test more robust, and also cleans up after itself now.

File: example/spec/parallel/testAPI/snsRuleSpec.js [hunk 2]
```diff
@@ -77,11 +88,18 @@ describe('The SNS-type rule', () => {
       expect(postRule.record.state).toEqual('ENABLED');
     });
 
-    it('creates a subscription and policy when it is created in an enabled state', async () => {
+    it('creates a subscription when it is created in an enabled state', async () => {
```
Comment (Jkovarik) [inline example/spec/parallel/testAPI/snsRuleSpec.js]: 👍

File: example/spec/parallel/testAPI/snsRuleSpec.js (hunk 1 above)
Comment (Jkovarik) [reply example/spec/parallel/testAPI/snsRuleSpec.js]: 💯

File: packages/api/webpack.config.js [hunk 3]
```diff
@@ -1,26 +1,36 @@
...
-if(process.env.PRODUCTION) {
-  mode = 'production',
-  devtool = false  
-}
+'use strict';
 
 module.exports = {
-  mode,
-  entry: './index.js',
+  mode: process.env.PRODUCTION ? 'production' : 'development',
```
Comment (Jkovarik) [inline packages/api/webpack.config.js]: 👍

File: packages/api/config/lambdas.yml [hunk 4]
```diff
@@ -156,35 +157,35 @@ CreateReconciliationReport:
...
+  handler: index.handler
+  source: 'node_modules/@cumulus/api/dist/bulkDelete/'
 
 KinesisInboundEventLogger:
-  handler: index.logger
+  handler: index.handler
   timeout: 300
-  memory: 512
-  source: 'node_modules/@cumulus/api/dist/'
+  memory: 128
```
Comment (Jkovarik) [inline packages/api/config/lambdas.yml]: What's the actual upper limit on inbound kinesis ingest records?     

*edit* answered that myself, it's small enough that shouldn't be a big deal..... however why is the OutboundEventLogger not similarly sized?

I want to be a little cautious about these as they're not backed by a DLQ (it's compliance logging, but still....)

Comment (yjpa7145) [reply packages/api/config/lambdas.yml]: When I adjusted these, I based the memory allocation on how much memory each Lambda used during our integration tests.  We apparently invoke the InboundEventLogger during our tests, but not the OutboutEventLogger, so I adjusted the Inbound one but left Outbound the way it was.

**Pull request merged**